Hi,

On Tue, 31 Jan 2006, Christian T. Steigies wrote:

> ide: Assuming 50MHz system bus speed for PIO modes; override with idebus=xx
> ide0: Catweasel IDE interface
> ide1: Catweasel IDE interface
> ide2: Catweasel IDE interface
> hda: FUJITSU MPF3204AT, ATA DISK drive
> Unable to handle kernel access at virtual address 79735f7a

This might need the same fix, I already needed for gayle.c.

bye, Roman


Index: drivers/ide/legacy/buddha.c
===================================================================
RCS file: /home/linux-m68k/cvsroot/linux/drivers/ide/legacy/buddha.c,v
retrieving revision 1.2
diff -u -p -r1.2 buddha.c
--- linux/drivers/ide/legacy/buddha.c	7 Dec 2003 10:25:18 -0000	1.2
+++ linux/drivers/ide/legacy/buddha.c	1 Feb 2006 14:57:35 -0000
@@ -197,6 +197,7 @@ fail_base2:
 			z_writeb(0, buddha_board+BUDDHA_IRQ_MR);
 		
 		for(i=0;i<buddha_num_hwifs;i++) {
+			memset(&hw, 0, sizeof(hw));
 			if(type != BOARD_XSURF) {
 				ide_setup_ports(&hw, (buddha_board+buddha_bases[i]),
 						buddha_offsets, 0,

